src/main/java/com/atlassian/httpclient/apache/httpcomponents/ApacheAsyncHttpClient.java

```diff
@@ -137,121 +137,128 @@ public ApacheAsyncHttpClient(final Supplier<String> applicationName,
-                    .setDefaultIOReactorConfig(reactorConfig)
-                    .setConnectionManager(connectionManager)
-                    .setRedirectStrategy(new RedirectStrategy())
-                    .setUserAgent(getUserAgent(options))
-                    .setDefaultRequestConfig(requestConfig);
-
-            if(Jenkins.getInstance() != null) {
-                ProxyConfiguration proxyConfiguration = Jenkins.getInstance().proxy;
-                if ( proxyConfiguration != null ) {
-                    final HttpHost proxy = new HttpHost( proxyConfiguration.name, proxyConfiguration.port );
-                    //clientBuilder.setProxy( proxy );
-                    if ( StringUtils.isNotBlank( proxyConfiguration.getUserName() ) ) {
-                        clientBuilder.setProxyAuthenticationStrategy( ProxyAuthenticationStrategy.INSTANCE );
-                        CredentialsProvider credsProvider = new BasicCredentialsProvider();
-                        credsProvider.setCredentials( new AuthScope( proxyConfiguration.name, proxyConfiguration.port ),
-                                                      new UsernamePasswordCredentials( proxyConfiguration.getUserName(),
-                                                                                       proxyConfiguration.getPassword() ) );
-                        clientBuilder.setDefaultCredentialsProvider( credsProvider );
-                    }
+        final RequestConfig requestConfig = RequestConfig.custom()
+                .setConnectTimeout((int) options.getConnectionTimeout())
+                .setConnectionRequestTimeout((int) options.getLeaseTimeout())
+                .setCookieSpec(options.getIgnoreCookies() ? CookieSpecs.IGNORE_COOKIES : CookieSpecs.DEFAULT)
+                .setSocketTimeout((int) options.getSocketTimeout())
+                .build();
 
-                    clientBuilder.setRoutePlanner(
-                        new JenkinsProxyRoutePlanner( proxy, proxyConfiguration.getNoProxyHostPatterns() ) );
+        final HttpAsyncClientBuilder clientBuilder = HttpAsyncClients.custom()
+                .setThreadFactory(ThreadFactories.namedThreadFactory(options.getThreadPrefix() + "-io", ThreadFactories.Type.DAEMON))
+                .setDefaultIOReactorConfig(reactorConfig)
+                .setConnectionManager(connectionManager)
+                .setRedirectStrategy(new RedirectStrategy())
+                .setUserAgent(getUserAgent(options))
+                .setDefaultRequestConfig(requestConfig);
+
+        if(Jenkins.getInstance() != null) {
+            ProxyConfiguration proxyConfiguration = Jenkins.getInstance().proxy;
+            if ( proxyConfiguration != null ) {
+                final HttpHost proxy = new HttpHost( proxyConfiguration.name, proxyConfiguration.port );
+                //clientBuilder.setProxy( proxy );
```

Comment:
I just noted this (which results of #154) - isn't it an unintentional change that actually disables proxy usage at all? 
maybe @escoem remembers that..